docs/OnTrack/Deployment/Enhanced Authentication/current-and-proposed-authentication-evaluation-5.1.md

```diff
@@ -0,0 +1,185 @@
+displayed – the token is cleared from the browser, however, so it still works in that regard.
+
+A second risk is that this current method of authentication relies on third-party and outsourced
+authentication technologies. Through utilising the services provided by Deakin SSO and MFA
+facilitated by Duo Security, although these technologies may be highly efficient, secure and
+reliable, such a reliance on third-party software means that the backend workings of this software
+is not able to be accessed and understood by the OnTrack team. Additionally, use of third-party
+software requires additional sharing, transmission, and storage of user information on systems which
+are not able to be managed by the OnTrack team. In the case of a security breach, the OnTrack team
+are also forced to rely upon the providers of these technologies to:
+
+- Report that the breach happened
+- Report which details and systems have been compromised
+- Fix the issues which lead to the breach
+- Secure the system and continue normal business operations
+
+Having to rely upon other vendors for these processes removes the control and level which OnTrack is
+able to gain information about the scale and nature of security breaches. This leaves the company
+“in the dark” about what has occurred and is frustrating, especially when there are capable members
+within the company who would be able to respond to such events, perhaps more efficiently than these
+vendors.
+
+Hence, from these risks and issues associated with the current OnTrack authentication mechanisms, it
+is clear why new technologies and methods of authentication have been proposed to be developed and
+implemented.
+
+## Proposed Authentication Enhancements
+
+This section will detail the proposed authentication elements to be added to the OnTrack
+architecture in order to improve the state of the authentication mechanisms and to alleviate some of
+the risks and errors in the described current setup.
+
+### The Proposed Authentication Mechanism
+
+The solution which has been proposed to be implemented by the Enhance Authentication team has
+several elements, as follows:
+
+- An extension of the current OnTrack API, adding code which will facilitate user management
+  functionalities
+- In line with this API extension, the addition of a feature to allow user passwords to be changed,
+  either by the user themselves or through sending a request to the admin to authorise the password
+  change. It is important that, as part of this feature, the admin is able to manipulate the
+  password (i.e. send the request for a password change) but not allow to see the plaintext password
+  at any time during this process, both before and after the password change
+- The addition of a Devise LDAP server option which handles the authentication processes for
+  OnTrack, allowing the authentication to be performed fully “in-house” rather than outsourced to
+  other authentication mechanisms
```

Comment:
Nitpick: reword to simplify. (Might need formatting if you add this suggestion)

```suggestion
- An extension of the current OnTrack API, adding functionality to facilitate user management
- Password management features, for users and admin manipulation. Users can either send
a request to the admin to authorise the password or do so themselves. It is important that,
as part of this feature, the admin is able to manipulate the password (i.e. send the request
for a password change). The password is never transferred in plaintext at any time during
this process.
- A Devise LDAP server option which handles the authentication processes for OnTrack,
allowing the authentication to be performed fully “in-house” rather than outsourced to other
authentication mechanisms
```